### ISSUE TITLE
Connect and show logon screen

### ISSUE THREAD

**Issue by windylea** [cedrozor/myrtille#116]
I have installed Myrtille on Windows 10 and everything works great.

However, is it possible for Myrtille to connect and show the logon screen? I'm on a monitored network that even SSL traffic is intercepted and I fear that someone could sniff out the username and password. If I get access to the logon screen, I could use the on-screen keyboard to type and avoid the risk. Thanks.

**Comment by cedrozor** [comment]
It should be possible yes, but implies to disable NLA on the target server: https://serverfault.com/questions/885080/is-it-possible-to-access-the-login-windows-screen-through-rdp

Be cautious with it as it will bring other security risks such as DoS attacks.

You can enable the enterprise mode of myrtille to be able to select the security mode of the connection (None, RDP, TLS, NLA); I think I will make that choice available by default into the next myrtille versions because there were recently some MS updates regarding NLA and CredSSP which disrupted FreeRDP and even mstsc on out of date machines.

**Comment by cedrozor** [comment]
I just (re)tested and it works; First uncheck "enforce NLA" on the target server:

![](http://2.bp.blogspot.com/-f0ptXhFCdP0/UzD0Nb70NuI/AAAAAAABj4s/fPNrQLKyHI0/s1600/2014-03-24+10-49-42+PM.png)

Then it requires myrtille to use RDP or TLS security mode for the host connection, with empty username and password, so that you are displayed the Windows logon screen.

Unfortunally, contrary to what I said, the [myrtille enterprise mode](https://github.com/cedrozor/myrtille/blob/master/DOCUMENTATION.md#enterprise-mode), won't help because when you connect a host from your dashboard, it will use your login credentials to connect it, and they are not empty.

The "Create Single Use Session URL" won't work either because it also requires a username and password.

This is frustrating :/ so, waiting for the next release and how I fine tune this issue (I think I will add the security mode selection into the default login screen), you can use the attached **Myrtille.Services.exe** ; it will look for the *FreeRDPSecurityMode* key into **Myrtille.Services.exe.config**, so you can set **"FreeRDPSecurityMode=rdp"**.

[Myrtille.Services.SecurityMode.Patch.1.9.x.zip](https://github.com/cedrozor/myrtille/files/2127881/Myrtille.Services.SecurityMode.Patch.1.9.x.zip)

This little out of track patch is meant for *Myrtille 1.9.x*

To summarize:
- stop the Myrtille.Services Windows service
- replace Myrtille.Services.exe into the myrtille bin folder by the one attached to this post
- restart Myrtille.Services
- ensure your target server doesn't enforce NLA (see above picture)
- into the myrtille login screen, enter the target address and leave username and password empty

**Comment by bigpjo** [comment]
If someone is decrypting and monitoring all traffic using an on screen keyboard will not help as the https traffic can also be monitored to intercept the credentials. In order to decrypt the traffic (using wiresshark for argument sake) the private key for ssl cert will be required. As an opinion for remote access it would be far more important to request credentials to connect than to allow rdp connection without credentials and use standard windows login.

**Comment by cedrozor** [comment]
I think the idea was to use the on-screen keyboard within the remote
session, into the windows logon screen,  by using the accessibility button;
that way, the only things going through the network are mouse clicks and
display updates.

For further protection, perhaps the on screen keyboard position could be
moved, so that mouse clicks coordinates change everytime.

Indeed, NLA is recommended and should be safer... but it doesn't allow what
the OP is looking for.

Le 23 juin 2018 21:30, "Paul" <notifications@github.com> a écrit :

If someone is decrypting and monitoring all traffic using an on screen
keyboard will not help as the https traffic can also be monitored to
intercept the credentials. In order to decrypt the traffic (using
wiresshark for argument sake) the private key for ssl cert will be
required. As an opinion for remote access it would be far more important to
request credentials to connect than to allow rdp connection without
credentials and use standard windows login.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/cedrozor/myrtille/issues/116#issuecomment-399702814>,
or mute the thread

**Comment by bigpjo** [comment]
That makes sense. Sorry I misunderstood

**Comment by cedrozor** [comment]
No, you were right... the on screen keyboard shouldn't be run client side
(otherwise, keyboard inputs will be sent across the wire) ;)

Le 23 juin 2018 22:35, "Paul" <notifications@github.com> a écrit :

That makes sense. Sorry I misunderstood

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/cedrozor/myrtille/issues/116#issuecomment-399709134>,
or mute the thread

**Comment by windylea** [comment]
Yeah that's the point, the on-screen keyboard should be opened on the remote side. Also thank you for the hot fix but it doesn't work for me. Here is the log contents:

```
2018-06-24 11:50:00,182 [8] INFO  System.Diagnostics redirection [(null)] - Connecting remote session 1, server localhost, domain (none), user ****, program (none)
2018-06-24 11:50:00,354 [8] INFO  System.Diagnostics redirection [(null)] - Disconnected remote session 1, exit code 131084
```

The error message, which equivalents to exit code `131084`, is `The remote connection failed due to invalid or unreachable server address`. If I undo the fix and use current version file then everything is okay though. So.. what have I done wrong?

**Comment by windylea** [comment]
I have already enabled that settings at first place but still couldn't get things work. Interestingly, after searching for a while, I found out that it is no need to apply the patch for Myrtille, but I have to change this registry key `Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Terminal Server\WinStations\RDP-Tcp\SecurityLayer` to `0` 

[SecurityLayer - Windows 10 hardware dev](https://msdn.microsoft.com/en-us/library/windows/hardware/dn965839(v=vs.85).aspx "SecurityLayer - Windows 10 hardware dev")

So I guess the issue is fixed now. Thank you very much for your help too.

**Comment by cedrozor** [comment]
I tested under a Windows Server 2012 R2 and just needed to uncheck the NLA box; that's weird it's not enough for Windows 10 because it's the same checkbox... unless you have a GPO or something that overrides it?

Anyway, the registry entry is good enough; in fact I had it, among other entries, into [RDPSetup.reg](https://github.com/cedrozor/myrtille/blob/master/Myrtille.Services/RDPSetup.reg#L20) which was used into previouses versions of myrtille for automatic rdp servers configuration (and could still be used... proof is).

While you are here, there are some entries that could interest you, such as `MaxIdleTime`, `DisableCursorBlink`, etc.

**Comment by cedrozor** [comment]
Glad it's working anyway :) I'm not closing the issue for now as it may prove useful to people with the same needs.
